arangod/Aql/OptimizerRule.h

```diff
@@ -370,6 +370,9 @@ struct OptimizerRule {
     // for it.
     joinIndexNodesRule,
 
+    // Should be last search rule!
```

Comment:
Why last search?  As from rule it is fine to run it anytime as SEARCH condition is not being altered anymore. And here  `static_assert` below is actually valid.

Reply:
Because if someone will touch search condition after applying this rule, rule will be wrong

Reply:
Yes. But for example, nothing prevents from running search late materialization rule after this one.